src/main/java/ca/nrc/cadc/beacon/web/resources/PackageServerResource.java

```diff
@@ -0,0 +1,207 @@
+    }
+
+    @Get("json")
+    public Representation notSupported() throws Exception {
+        return new JSONRepresentation() {
+            @Override
+            public void write(final JSONWriter jsonWriter)
+                throws JSONException {
+                jsonWriter.object()
+                    .key("msg").value("GET not supported.")
+                    .endObject();
+            }
+        };
+    }
+
+    @Post("json")
+    public Representation getPackage(final JsonRepresentation payload) throws Exception {
+        final JSONObject jsonObject = payload.getJsonObject();
+        log.debug("getPackage input: " + jsonObject);
+
+        List<URI> targetList = new ArrayList<>();
+        final Set<String> keySet = jsonObject.keySet();
+
+        String responseFormat;
+        if (keySet.contains("responseformat")) {
+            responseFormat = jsonObject.getString("responseformat");
+        } else {
+            // default response format
+            responseFormat = "application/zip";
+        }
+
+        if (!keySet.contains("target")) {
+            return new JSONRepresentation() {
+                @Override
+                public void write(final JSONWriter jsonWriter)
+                    throws JSONException {
+                    jsonWriter.object()
+                        .key("msg").value("no targets found.")
+                        .endObject();
+                }
+            };
+        } else {
+            final String targetStr = (String) jsonObject.get("target");
+            final String[] targets = targetStr.split(",");
+
+            // build target list to add to transfer
+            for (final String target : targets) {
+                // URIs are being passed in from the UI
+                URI targetURI = new URI(target);
+                log.debug("adding URI to transfer target list: " + targetURI.toString());
+                targetList.add(targetURI);
+            }
+
+            // Create the Transfer.
+            Transfer transfer = new Transfer(Direction.pullFromVoSpace);
+            transfer.getTargets().addAll(targetList);
+
+            List<Protocol> protocols = new ArrayList<Protocol>();
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTP_GET));
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTPS_GET));
+            transfer.getProtocols().addAll(protocols);
+
+            // Add package view for tar file
+            View packageView = new View(new URI(Standards.PKG_10.toString()));
+            packageView.getParameters().add(new View.Parameter(new URI(VOS.PROPERTY_URI_FORMAT), responseFormat));
+            transfer.setView(packageView);
+
+            transfer.version = VOS.VOSPACE_21;
+
+            final ClientTransfer ct = voSpaceClient.createTransfer(transfer);
+            URL packageURL = new URL(ct.getTransfer().getProtocols().get(0).getEndpoint());
+
+            String endpoint = packageURL.toString();
+            if (endpoint != "") {
```

Comment:
A URL object cannot exist with an empty string.  I think the URL will either be null or non existent won't it?

Reply:
This conversion to a URL wasn't needed. Reworked this section so the endpoint is checked for length (StringUtil.hasLength) - which should check for null as well.